apps/api/src/campaign/campaign.service.ts

```diff
@@ -713,8 +898,111 @@ export class CampaignService {
+    // Create a notification list for this campaign when updated to active
+    if (
+      campaign?.state !== CampaignState.active &&
+      updated?.state === CampaignState.active &&
+      // No list exists yet
+      !campaign?.notificationLists?.length
+    )
+      try {
+        await this.createCampaignNotificationList(updated)
+
+        // Notify for new activated campaign
+        //Do not await -> send in background
+        this.sendNewCampaignNotification(updated).catch((e) => console.log(e))
+      } catch (e) {
+        Logger.error('Failed to create notification list', e)
+      }
+
+    // Update notification list name with the new campaign name
+    if (
+      updated?.state === CampaignState.active &&
+      campaign?.notificationLists?.length &&
+      updated.title !== campaign.notificationLists[0]?.name
+    )
+      try {
+        // Get the list
+        const list = campaign?.notificationLists[0]
+        if (list) await this.updateCampaignNotificationList(updated, list)
+      } catch (e) {
+        Logger.error('Failed to update notification list', e)
+      }
+
+    return updated
+  }
+
+  async createCampaignNotificationList(updated: { title: string; id: string }) {
+    // Generate list in the marketing platform
+    const listId = await this.marketingNotificationsService.provider.createNewContactList({
+      name: updated.title || updated.id,
+    })
+    // Save the list_id in the DB
+    await this.prisma.notificationList.create({
+      data: {
+        id: listId,
+        name: updated.title,
+        campaignId: updated.id,
+      },
+    })
+
+    return listId
+  }
+
+  async sendNewCampaignNotification(campaign: Campaign) {
+    // Send notification for the new activated campaign
+    const template = await this.prisma.marketingTemplates.findFirst({
+      where: {
+        name: 'New Active Campaign',
+      },
+    })
+
+    // General marketing list
+    const mainList = this.config.get('sendgrid.marketingListId')
+
+    if (template) {
+      const data: SendGridParams['SendNotificationParams'] = {
+        template_id: template.id,
+        list_ids: [mainList],
+        subject: NotificationData['new-campaign'].subject,
+        template_data: {
+          'campaign.name': campaign?.title,
+          'campaign.target-amount': campaign?.targetAmount || 0,
+          'campaign.start-date': campaign.startDate
+            ? DateTime.fromJSDate(campaign.startDate).toFormat('dd-MM-yyyy')
+            : '',
+          'campaign.end-date': campaign.endDate
+            ? DateTime.fromJSDate(campaign.endDate).toFormat('dd-MM-yyyy')
+            : '',
+          'campaign.link': (
+            this.config.get<string>('APP_URL') + `/campaigns/${campaign.slug}`
+          ).replace(/(http:\/\/|https:\/\/)/gi, ''),
+          'campaign.news-link': (
+            this.config.get<string>('APP_URL') + `/campaigns/${campaign.slug}/news`
+          ).replace(/(http:\/\/|https:\/\/)/gi, ''),
+        },
+      }
```

Comment:
That's a good idea and also it's a bit tricky, because every notification template would require different fields and probably different data sources( campaigns, vaults, people, news etc.). We're only creating the object here -. if we create an utility function it would probably take an object of all the values to build the data object itself, which I think wouldn't make the code leaner